src/writer/mod.rs

```diff
@@ -125,6 +132,27 @@ pub trait Failure<World>: Writer<World> {
     fn hook_errors(&self) -> usize;
 }
 
+/// [`Writer`] tracking a number of [`Passed`] or [`Skipped`] [`Step`]s.
+///
+/// [`Passed`]: event::Step::Passed
+/// [`Skipped`]: event::Step::Skipped
+/// [`Step`]: gherkin::Step
+pub trait SuccessOrSkipped<World>: Writer<World> {
```

Comment:
@ilslv looking at the usage, I don't see any reason to keep this separately from `writer::Failure` crate. Seems like merging them into a single `writer::Stats` would be just enough and much more understandable.

Reply:
And don't forget to rename `writer::discard::Failure` too.